src/scripts/main.js

```diff
@@ -1 +1,78 @@
+/* eslint-disable no-shadow */
 'use strict';
+
+document.addEventListener('DOMContentLoaded', () => {
+  let leftClicked = false;
+  let rightClicked = false;
+
+  const firstPromise = new Promise((resolve, reject) => {
+    const handleLeftClick = (event) => {
+      if (event.button === 0) {
+        resolve('First promise was resolved');
+        document.removeEventListener('mousedown', handleLeftClick);
+      }
+    };
+
+    document.addEventListener('mousedown', handleLeftClick);
+
+    setTimeout(() => {
+      reject(new Error('First promise was rejected'));
+      document.removeEventListener('mousedown', handleLeftClick);
+    }, 3000);
+  });
+
+  const secondPromise = new Promise((resolve) => {
+    const handleClick = (event) => {
+      if (event.button === 0 || event.button === 2) {
+        resolve('Second promise was resolved');
+        document.removeEventListener('mousedown', handleClick);
+      }
+    };
+
+    document.addEventListener('mousedown', handleClick);
+  });
+
+  const thirdPromise = new Promise((resolve) => {
+    const checkBothClicks = () => {
+      if (leftClicked && rightClicked) {
+        resolve('Third promise was resolved');
+        document.removeEventListener('mousedown', handleClick);
```

Comment:
The event listener 'handleClick' is removed here, but it is not the same 'handleClick' function used in the thirdPromise. This could lead to unexpected behavior as the listener is not properly removed. Consider using the same function reference for adding and removing event listeners.